src/shared/sleep-config.c

```diff
@@ -165,8 +165,30 @@ int can_sleep_disk(char **types) {
 
 #define HIBERNATION_SWAP_THRESHOLD 0.98
 
+/* entry in /proc/swaps */
+typedef struct SwapEntry {
+        char *device;
+        char *type;
+        uint64_t size;
+        uint64_t used;
+        int priority;
+} SwapEntry;
+
+static SwapEntry* swap_entry_free(SwapEntry *se) {
+        if (se) {
```

Comment:
doesn't matter at all, but I am a big fan of minimizing nesting, and early exit, so i would normally just have done this:

```c
if (!se)
        return NULL;
…
```